sphinxcontrib/confluencebuilder/builder.py

```diff
@@ -148,10 +150,32 @@ def get_relative_uri(self, from_, to, typ=None):
                             self.get_target_uri(to, typ))
 
     def prepare_writing(self, docnames):
+        ordered_docnames = []
+        traversed = [self.config.master_doc]
+
+        # prepare caching doctree hook
+        #
+        # We'll temporarily override the environment's 'get_doctree' method to
+        # allow this extension to manipulate the doctree for a document inside
+        # the pre-writing stage to also take effect in the writing stage.
+        self._original_get_doctree = self.env.get_doctree
+        self.env.get_doctree = self._get_doctree
+
+        # process the document structure of the master document, allowing:
+        #  - populating a publish order to ensure parent pages are created first
+        #     (when using hierarchy mode)
+        #  - squash pages which exceed maximum depth (if configured with a max
+        #     depth value)
+        self.process_tree_structure(
+            ordered_docnames, self.config.master_doc, traversed)
+
+        # add orphans (if any) to the publish list
+        ordered_docnames.extend(x for x in docnames if x not in traversed)
```

Comment:
Next time you can use a filter instead, which would return a generator in Python 3 and be slightly more efficient.